### PR TITLE
Bootstrap5

### PR DESCRIPTION
**Pull request by zebleck**
| 📆    | Merge of this branch scheduled for the 14th of June 2023, 10:00pm |
|---------------|:-------------------------|

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Branch for upgrade from bootstrap 4 to bootstrap 5.

**Official link for Bootstrap migration:**
https://getbootstrap.com/docs/5.0/migration/

Generally tried to recreate behavior and look as it was in v4.

**What still needs to be addressed:**

The hack for bootstrap accordions, that we use in v4, is not yet removed in v5.

Thorough testing probably required again (last time was before a lot of fixes).

**What has been changed:**

Some of the details of the changes (taken from my local documentation):

- reintroduced the href underlines since they are not underlined in bootstrap 5 by default
- helpdesks migrated
- lots of renaming things:
  - form-checkbox -> form-check
  - custom-control-input -> form-check-input
  - badge badge-light -> badge badge-light text-dark
  - “custom-control custom-radio custom-control-inline” -> “form-check form-check-inline”
  - data-toggle=”pill” -> data-bs-toggle=”pill”
  - .pl-* and .pr-* -> .ps-* and .pe-*.
  - .border-left and .border-right -> .border-start and .border-end.
  - .text-left and .text-right -> .text-start and .text-end.
- Navbars now require a container within (to drastically simplify spacing requirements and CSS required).
- fixed accordions (for profile/subscriptions, lecture/form and start page)
- Remove form-group, replace with form-floating mb-3
- Jumbotron removed, has to be replaced with other utilities

Fixed issues as are outlined in basecamp: https://3.basecamp.com/3978947/buckets/7198402/todos/5678353440#__recording_5887260341

and more changes as are outlined in the official link for migrating v4 to v5.

**To migrate an accordion**:
change “card” to “accordion-item”
change “div class=”card-header clickable”” zu “button class=”accordion-button collapsed””
change “data-*” to “data-bs-*”
change “card-body” to “accordion-body”

### CODE REVIEW

**Comment by Splines** [comment]
Thanks for opening this PR ;) Will look into it soon.

File: docker/run_cypress_tests/Dockerfile [hunk 1]
```diff
@@ -31,5 +31,8 @@ RUN bundle install && \
 COPY --from=build-pdfcomprezzor /go/src/pdfcomprezzor.wasm /go/src/wasm_exec.js /usr/src/app/public/pdfcomprezzor/
 COPY --from=build-pdfcomprezzor /go/src/pdfcomprezzor.wasm /go/src/wasm_exec.js /
 
+COPY Gemfile Gemfile.lock package.json yarn.lock /usr/src/app/
```
Comment (fosterfarrell9) [inline docker/run_cypress_tests/Dockerfile:34]: Is that related to the PR?

File: spec/cypress/e2e/simple_spec.cy.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+describe('My First Test', function() {
```
Comment (fosterfarrell9) [inline spec/cypress/e2e/simple_spec.cy.js:1]: Is that related to the PR?

File: spec/cypress/e2e/testing_spec.cy.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            cy.get('input[type="submit"]').click();
+        });
+		it("can view tutorials", () => {
+            cy.appFactories([
+            ]).then((lectures) => {
+                
+            });
+        });
+    });
+});
```
Comment (fosterfarrell9) [inline spec/cypress/e2e/testing_spec.cy.js:22]: Is that related to the PR?

**Review by fosterfarrell9 — CHANGES_REQUESTED** [review]
There are still some occurences of the `custom-control` css class for checkboxes (probably more than I marked, I noticed them at some point during the review). I have not found this class in the Bootstrap 5 documentation (it was however used in Bootstrap 4).

File: app/views/media/_publish_modal.html.erb [hunk 4]
```diff
@@ -69,29 +66,31 @@
             <div class="mb-1">
               <%= t('basics.assignment') %>
             </div>
-            <div class="custom-control custom-checkbox mb-2">
+            <div class="custom-control form-check mb-2">
```
[the remark this answers is not in the document]
Comment (Splines) [reply app/views/media/_publish_modal.html.erb]: Oh, glad you noticed this, replaced it now.

File: app/assets/javascripts/media.coffee [hunk 5]
```diff
@@ -389,11 +395,9 @@ $(document).on 'turbolinks:load', ->
     return
 
   $('#release_date').on 'focus', ->
+    console.log('focussss')
```
[the remark this answers is not in the document]
Comment (Splines) [reply app/assets/javascripts/media.coffee:398]: yes, removed

File: spec/cypress/e2e/simple_spec.cy.js (hunk 2 above)
Comment (Splines) [reply spec/cypress/e2e/simple_spec.cy.js:1]: No, this is not at all related to this PR. By doing a

```
git log -p -- ./spec/cypress/e2e/simple_spec.cy.js
```

from the current branch, I found out that the file was introduced when I merged `mampf-next` into this branch earlier on, see [here](https://github.com/MaMpf-HD/mampf/commit/843854fe8317103b12b78e903a10ae17d59de77c#diff-f971a7862ba60ddb68800528aa4209773ddb3622a1507a123b0afc5aa3271a98). The file was renamed, so it popped up here as well. However, meanwhile I cannot find the file in the `mampf-next` branch anymore which is a bit strange as `git merge mampf-next` just gives me an "Already up to date".

Found out that the file was deleted on mampf-next [in this commit](https://github.com/MaMpf-HD/mampf/commit/a0557b868b397d85d72fea4d5b73566c192393a7#diff-f971a7862ba60ddb68800528aa4209773ddb3622a1507a123b0afc5aa3271a98) by running this command (from the mampf-next branch):
```
git rev-list -n 1 HEAD -- ./spec/cypress/e2e/simple_spec.cy.js
```

This is all a bit weird. To replay my merge and find out why the file was added there I did a:
```
git checkout 843854^
git merge mampf-next
```

I faced merge conflicts again but the `simple_spec.cy.js` file was not touched. So in short, I don't know why I readded this file in my merge commit. I will just delete it as not needed.

File: spec/cypress/e2e/testing_spec.cy.js (hunk 3 above)
Comment (Splines) [reply spec/cypress/e2e/testing_spec.cy.js:22]: no, same problem as with `simple_spec.cy.js`. Will remove.

File: docker/run_cypress_tests/Dockerfile (hunk 1 above)
Comment (Splines) [reply docker/run_cypress_tests/Dockerfile]: nope, removed

**Review by fosterfarrell9 — APPROVED** [review]

**Review by fosterfarrell9 — APPROVED** [review]
